From cf39f5ea277a226a4f102065a030f261f89c42af Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Thu, 13 Aug 2015 18:47:16 -0500
Subject: [PATCH] skb_recycler: always set skb->dev

Without this, if we call skb_recycler_alloc with
dev == NULL, the pointer in skb->dev could be stale,
so we just always set skb->dev instead

Change-Id: I32670c64a3932239aa0758acc9cd6482fd07d053
Signed-off-by: Matthew McClintock <mmcclint@codeaurora.org>
Signed-off-by: Casey Chen <kexinc@codeaurora.org>
---
 net/core/skbuff_recycle.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/net/core/skbuff_recycle.c
+++ b/net/core/skbuff_recycle.c
@@ -88,9 +88,7 @@ inline struct sk_buff *skb_recycler_allo
 		skb->data = skb->head + NET_SKB_PAD;
 		skb_reset_tail_pointer(skb);
 
-		if (dev)
-			skb->dev = dev;
-
+		skb->dev = dev;
 		skbuff_debugobj_activate(skb);
 	}
 
